### ISSUE TITLE
Feature Request: Allow setting individual ownership of separate but duplicate items

### ISSUE THREAD

**Issue by doddsy** [grocy/grocy#418]
Hi there, I hope I'm doing this in the right format.

I encountered Grocy yesterday afternoon and after briefly pitching it to my flatmates we decided to give it a go. It's a really great system but there's one thing missing for us, which is the ability to set who owns one of an item.

For example, say Flatmate A buys eggs, enters all the pertinent info, nothing wrong there. But then Flatmate B also decides to buy eggs, enters all the pertinent info (with separate Best Before dates, locations, etc.) - Grocy will then group those items together even though they have separate info.

I was hoping that with separate Location settings per item, and then sorting the stock overview by an individual location, I could trick it into only displaying items in that user's cupboard/fridge/wherever, however it still groups the items together as if they were the same.

If there's a workaround for this that already exists, please let me know as I couldn't find one. But I still feel like this would be a great addition to Grocy, especially for those in flats or shared housing.

**Comment by berrnd** [comment]
Great idea! :)

The only other workaround would maybe to create subproducts per "owner", although not really ideal...

**Comment by doddsy** [comment]
Yeah, I had considered doing that but that definitely would've been confusing for the others and a bit clunky to work around.

**Comment by duylong** [comment]
Good feature, related also to this request #827

I did not find a simple solution to manage the common products and belonging to other people. For now I have to test it alone, do you have any suggestions to work around the problem?

**Comment by firefloris** [comment]
I would really like this, I'm too living with flatmates.
Grocy is kinda useless to me because of this, contrary to its awesome features.

**Comment by JessBugarin** [comment]
Adding that I'm a seeking this as well.  For my use case, the Stock entry needs to be ownable by more than one user (but not all) as well as optionally by all users.

Example: 5 housemates. A carton of eggs is owned by housemates A, B & C. A different carton of eggs is owned by housemate D.  But this can of beans might be owned by housemates A & B only and not C.  Some food is shared, some is not. Which food is shared varies. But knowing who's food is whose, and what is ok to consume, is important. If I'm C, I want to know which eggs my housemates bought with my funds and I'm allowed to eat, and I want to know that the can of beans wasn't purchased with my money and isn't mine to consume.

I'm going to play with the Product Groups and Master Products to see if I can get one of these to play this role for now.
